src/pages/companion/profile/[externalUserId]/index.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { fetchBackend } from "@/lib/db";
+import PageError from "@/components/companion/PageError";
+import { Loader2, XCircleIcon } from "lucide-react";
+import { set } from "lodash";
+
+interface Registration {
+    id: string;
+    fname: string;
+    points?: number;
+    [key: string]: any;
+}
+
+interface Qr {
+    data: Record<string, any>;
+    id: string;
+    type: string;
+}
+
+const index = () => {
+    const eventID = "blueprint";
+    const year = "2025";
+
+    const router = useRouter();
+    const [qrData, setQrData] = useState<Qr | null>(null);
+    const [qrPath, setQrPath] = useState<string>();
+    const [localUser, setLocalUser] = useState<string>();
+    const [userId, setUserId] = useState("");
+    const [registrations, setRegistrations] = useState<Registration[]>([]);
+    const [loading, setLoading] = useState(true);
+    const [loadingQr, setQrLoading] = useState(true);
+    const [pageError, setPageError] = useState("");
+
+    // fetching user data from qr code
+    const fetchUserID = async (qrId: string) => {
+        try {
+            console.log(qrId);
+            const response = await fetchBackend({
+                endpoint: `/qr/${qrId}/${eventID}/${year}`,
+                method: "GET",
+                authenticatedCall: false,
+            });
+
+            const res = await response;
+            if (res) {
+                setQrData(res);
+                setUserId(res.data.registrationID);
+            } else if (qrPath) {
+                setPageError("NFC data not found. Try scanning your NFC card again.");
+            }
+            setQrLoading(false);
+        } catch (err) {
+            setPageError(
+                (err as Error)?.message || "An unexpected error occurred"
+            );
+        }
+    };
+
+    // fetching registrations data
+    const fetchRegistrations = async () => {
+        try {
+            const response = await fetchBackend({
+                endpoint: `/registrations?eventID=${eventID}&year=${year}`,
+                method: "GET",
+                authenticatedCall: false,
+            });
```

Comment:
ignore - Adding a reminder here for myself, will need to modify this to use the new Companion Registrations API